airflow/providers/amazon/aws/hooks/redshift_statement.py

```diff
@@ -0,0 +1,144 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Interact with AWS Redshift, using the boto3 library."""
+
+from typing import Callable, Dict, Optional, Tuple, Union
+
+import redshift_connector
+from redshift_connector import Connection as RedshiftConnection
+
+from airflow.hooks.dbapi import DbApiHook
+
+
+class RedshiftStatementHook(DbApiHook):
```

Comment:
I agree with this.

Reply:
Love the idea. Personally I'd vote for renaming the existing hook to `RedshiftClusterHook`.

Getting to a point where users don't need to cross-pollinate their environment with the Postgres provider just to simply execute SQL in Redshift would be very beneficial. I agree that transition will take some time but it most certainly will be worth the effort.

Reply:
Would it be possible to include the hook under its current name, `RedshiftStatementHook`, so we don't need to wait for 3.0, and then add in some shim for backwards compatibility in the future? 

i.e.
```python
RedshiftHook = RedshiftStatementHook
```

While RedshiftOperator would likely work with the Postgres hook, it would be lacking the primary feature requests in #16355, which include extended datatype support and authentication via IAM and Identity provider. 

IMO, the authentication bits are especially important because they simplify an existing workflow (connecting to Redshift with temporary credentials), and open this integration up to a new group of users (those requiring authentication with Redshift via an identity provider (e.g. Okta, Azure)).

Reply:
What if we just call the module redshift_sql but the hook Redshift hook

Then there are two redshift hooks until 3.0

After 3.0 the old one is renamed

Reply:
that sounds good to me, I'm happy to make this change if the others here agree with taking this route @josh-fell @JavierLopezT

Reply:
How about something like `RedshiftSqlHook` instead of having two hooks named `RedshiftHook`? Otherwise I feel it may be confusing to users and seems relevant to the features being built with this PR as well as keeping to naming conventions. Although, this suggestion would may warrant renaming the new operator to match.  

If `RedshiftSqlHook` is agreeable then maybe we don't need to rename it later either.

Reply:
I think RedshiftSqlHook is an acceptable name, though I reached out on slack to see if anyone has some advice for how we could end up with `RedshiftHook` as the sql hook.

Reply:
thank you for reaching out on slack, @dstandish. I think having two RedshiftHooks until 3.0, as you wrote on slack, would be a good path to getting this merged with consistent naming with the other database hooks.

Reply:
@mik-laj agreed with @josh-fell on this one, and i think i now agree with this too.

namely we should do the following:
* add RedshiftSqlHook to the existing `redshift` module (i.e. in the same module as the existing RedshiftHook)
* deprecate RedshiftHook (renaming it to RedshiftClusterHook) -- this doesn't necessarily need to happen in this PR

then in the next major release _for this provider_ (which now that i think of it does not have to be in 3.0, since providers have separate release schedule), we remove `RedshiftHook` and we are left with `RedshiftSqlHook` and `RedshiftClusterHook`

Reply:
This sounds good. As @josh-fell mentioned,
> Although, this suggestion would may warrant renaming the new operator to match.

We can rename the operator to `RedshiftSqlOperator` as well so the naming is inline with `RedshiftSqlHook`.

Reply:
> When using acronyms in CapWords, capitalize all the letters of the acronym. Thus HTTPServerError is better than HttpServerError.

[PEP-8](https://www.python.org/dev/peps/pep-0008/)

We should use `RedshiftSQLHook` instead of `RedshiftSqlHook`

Reply:
thanks to all for collaborating to find a good name for the hook and operator! :) 

I've made the rename to `RedshiftSQLHook` and `RedshiftSQLOperator`, and placed `RedshiftSQLHook` in the same module as `RedshiftHook` as Kamil mentioned on slack in a5e8786